java/vector/src/main/java/org/apache/arrow/vector/dictionary/Dictionary.java

```diff
@@ -52,14 +52,6 @@ public String toString() {
   }
 
   @Override
-  public boolean equals(Object o) {
```

Comment:
I don't think that it would affect anything. IMO seems better to leave it and add a ticket to fix the vector.equals, although maybe there is a reason vectors don't implement equals.